### PR TITLE
refactor: fix javadoc

### PR DESCRIPTION
**Pull request by zielint0**
@surli 
I corrected some Javadoc errors. Please review it deeply and suggest missing information. Thanks!

- ~~PatternParameterConfigurator.java -> I removed non-existing parameters. What is *pep*?~~
- ~~DefaultGenerator.java -> There is no class GeneratorImpl. Should it be a *Generator*?~~ -> yes
- ~~PatternPrinter.java -> I removed non-existing parameters. What is *parameterInfo*?~~
- ~~ElementNode.java -> *roleHandler* changed to *role*. Check full doc of this method please.~~
- ~~SingleHandler.java -> I removed non-existing *V*. Added *U*. Should I add one more *U*?~~

I am ready to merge.

### CODE REVIEW

File: src/main/java/spoon/reflect/meta/impl/SingleHandler.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 /**
  * implementation of {@link RoleHandler}, which handles attributes of type &lt;V&gt;
```
Comment (surli) [inline src/main/java/spoon/reflect/meta/impl/SingleHandler.java]: You missed a V here.

**Comment by surli** [comment]
> - PatternParameterConfigurator.java -> I removed non-existing parameters. What is pep?
>  - PatternPrinter.java -> I removed non-existing parameters. What is parameterInfo?

@pvojtechovsky could you answer for those ones?

>  - DefaultGenerator.java -> There is no class GeneratorImpl. Should it be a Generator?

yes

>  - ElementNode.java -> roleHandler changed to role. Check full doc of this method please.
>  - SingleHandler.java -> I removed non-existing V. Added U. Should I add one more U?

Not sure to understand the question: what do you mean about adding one more U?

**Comment by pvojtechovsky** [comment]
> PatternParameterConfigurator.java -> I removed non-existing parameters. What is pep?

@param pep pair of parameter and element which has to be transformed

> PatternPrinter.java -> I removed non-existing parameters. What is parameterInfo?

@param parameterInfo describes a to be printed pattern parameter 

> ElementNode.java -> roleHandler changed to role. Check full doc of this method please.

new comment is correct.

> SingleHandler.java -> I removed non-existing V. Added U. Should I add one more U?

No, one U is enough.

File: src/main/java/spoon/reflect/meta/impl/SingleHandler.java (hunk 1 above)
Comment (zielint0) [reply src/main/java/spoon/reflect/meta/impl/SingleHandler.java]: ?

**Comment by zielint0** [comment]
@pvojtechovsky 
Thank you for all explanations.

**Comment by zielint0** [comment]
What is `@param type` in spoon/pattern/internal/PatternPrinter.java 
`generatePatternParameterElement(ParameterInfo parameterInfo, Class<T> type)?`

**Comment by pvojtechovsky** [comment]
> What is @param type in spoon/pattern/internal/PatternPrinter.java

it is a type/class of to be generated element

**Comment by zielint0** [comment]
@pvojtechovsky 
Thank you!

File: src/main/java/spoon/reflect/meta/impl/SingleHandler.java (hunk 1 above)
Comment (surli) [reply src/main/java/spoon/reflect/meta/impl/SingleHandler.java]: It says that it handles attributes of type V but there is no type V defined in the class signature now.

Comment (zielint0) [reply src/main/java/spoon/reflect/meta/impl/SingleHandler.java]: Class signature is: abstract class SingleHandler<T, U> extends AbstractRoleHandler<T, U, U>
so why 'V' should be documented?

Comment (surli) [reply src/main/java/spoon/reflect/meta/impl/SingleHandler.java]: It should not but currently the comment is the following:
>[...] which handles attributes of type &lt;V&gt;

So V or `&lt;V&gt;` is currently documented and should not be.

Comment (zielint0) [reply src/main/java/spoon/reflect/meta/impl/SingleHandler.java]: Ok, not it is not documented.

**Pull request merged**